src/blockchain_db/blockchain_db.h

```diff
@@ -1464,12 +1485,12 @@ class BlockchainDB
   /**
    * @brief get the number of transactions in the txpool
    */
-  virtual uint64_t get_txpool_tx_count(bool include_unrelayed_txes = true) const = 0;
+  virtual uint64_t get_txpool_tx_count(relay_category tx_category = relay_category::broadcasted) const = 0;
 
   /**
    * @brief check whether a txid is in the txpool
    */
-  virtual bool txpool_has_tx(const crypto::hash &txid) const = 0;
+  virtual relay_method txpool_has_tx(const crypto::hash &txid) const = 0;
```

Comment:
I've become some kind of haskell purist. Out parameters, especially an enum out parameter, makes me squeamish. I had `boost::optional<relay_method>` at one point, but this seemed slightly better. I didn't think think it was a big deal because in C/C++ you have to handle the `default` case anyway since the value can be munged.

I'll revert to out parameter if you think its better from a consistency perspective on the enum values, but its not really going to simplify any of the calls. Most (all?) of the existing calls have to be more complicated to check the out parameter in some way. There a few cases where a `switch (...)` will become a multi-line setup too.

Reply:
I was thinking of all the changes to "if (txpool_has_tx)" calls. If you pass a relay_method*=NULL, none change, and you can get your data if you need it. It just feels wrong to me, but if others prefer it too, I guess it'd be ok (in that case, maybe renaming it get_txpool_something would be better to match the new return).

Reply:
I decided to make this change, it doesn't look too bad so far. Most of the `get_txpool_blob` stuff can now skip a LMDB lookup too.

Reply:
So the actual change was a bit different - the return value has been reverted to `bool` and the `relay_method::not_found` value has been too. Instead, this takes a `relay_category` argument which filters the return value (i.e. it is attempting to filter based on "split" mempool).